RecoParticleFlow/PFProducer/plugins/PFProducer.cc

```diff
@@ -333,3 +309,233 @@ void PFProducer::produce(Event& iEvent, const EventSetup& iSetup) {
+  desc.add<int>(
+      "goodPixelTrackDeadHcal_maxLost3Hit",
+      0);  // max missing outer hits for a track with 3 valid pixel layers (can set to -1 to reject all these tracks)
+  desc.add<int>("goodPixelTrackDeadHcal_maxLost4Hit",
+                1);  // max missing outer hits for a track with >= 4 valid pixel layers
+  desc.add<double>("goodPixelTrackDeadHcal_dxy", 0.02);  // [cm] abs(trackRef->dxy(primaryVertex_.position())) < X
+  desc.add<double>("goodPixelTrackDeadHcal_dz", 0.05);   // [cm] abs(trackRef->dz(primaryVertex_.position())) < X
+
+  // number of sigmas for neutral energy detection
+  desc.add<double>("pf_nsigma_ECAL", 0.0);
+  desc.add<double>("pf_nsigma_HCAL", 1.0);
+
+  // ECAL/HCAL PF cluster calibration : take it from global tag ?
+  desc.add<bool>("useCalibrationsFromDB", true);
+  desc.add<std::string>("calibrationsLabel", "");
+
+  // Post HF cleaning
+  desc.add<bool>("postHFCleaning", false);
+  {
+    edm::ParameterSetDescription psd_PFHFCleaning;
+    // Clean only objects with pt larger than this value
+    psd_PFHFCleaning.add<double>("minHFCleaningPt", 5.0);
+    // Clean only if the initial MET/sqrt(sumet) is larger than this value
+    psd_PFHFCleaning.add<double>("maxSignificance", 2.5);
+    // Clean only if the final MET/sqrt(sumet) is smaller than this value
+    psd_PFHFCleaning.add<double>("minSignificance", 2.5);
+    // Clean only if the significance reduction is larger than this value
+    psd_PFHFCleaning.add<double>("minSignificanceReduction", 1.4);
+    // Clean only if the MET and the to-be-cleaned object satisfy this DeltaPhi * Pt cut
+    // (the MET angular resoution is in 1/MET)
+    psd_PFHFCleaning.add<double>("maxDeltaPhiPt", 7.0);
+    // Clean only if the MET relative reduction from the to-be-cleaned object
+    // is larger than this value
+    psd_PFHFCleaning.add<double>("minDeltaMet", 0.4);  //
+    desc.add<edm::ParameterSetDescription>("PFHFCleaningParameters", psd_PFHFCleaning);
+  }
+
+  // Check HF cleaning
+  desc.add<std::vector<edm::InputTag>>("cleanedHF",
+                                       {
+                                           edm::InputTag("particleFlowRecHitHF", "Cleaned"),
+                                           edm::InputTag("particleFlowClusterHF", "Cleaned"),
+                                       });
+
+  // calibration parameters for HF:
+  desc.add<bool>("calibHF_use", false);
+  desc.add<std::vector<double>>("calibHF_eta_step", {0.0, 2.9, 3.0, 3.2, 4.2, 4.4, 4.6, 4.8, 5.2, 5.4});
+  desc.add<std::vector<double>>("calibHF_a_EMonly", {1., 1., 1., 1., 1., 1., 1., 1., 1., 1.});
+  desc.add<std::vector<double>>("calibHF_a_EMHAD", {1., 1., 1., 1., 1., 1., 1., 1., 1., 1.});
+  desc.add<std::vector<double>>("calibHF_b_HADonly", {1., 1., 1., 1., 1., 1., 1., 1., 1., 1.});
+  desc.add<std::vector<double>>("calibHF_b_EMHAD", {1., 1., 1., 1., 1., 1., 1., 1., 1., 1.});
+
+  descriptions.add("particleFlow", desc);
```

Comment:
it looks like only _cff file should be exposed to users for a consistent use.
For this case, since there were known use cases of ```particleFlow_cfi```, it is better to avoid confusion and name this differencely, e.g. use the default label ```pfProducer``` or ```particleFlowDefault```

Reply:
Now, particleFlow_cfi.py is being removed. I think I can use e.g. particleFlowDefault, but I am not so sure why it's preferred.

Reply:
As in the current state of the PR, the particleFlow_cfi.py is not removed, it is instead replaced by the auto-generated particleFlow_cfi.py file which will be incomplete if anyone tries to use it.
the main reason to rename this auto-generated file is to avoid possible confusion for someone who had it in their old configuration (outside of the official CMSSW code tree).

Reply:
@slava77 What do you mean here? For ConfDB parsing we'd need a cfi file with at least all top-level parameters...

Reply:
@Martin-Grunewald 
I expect that you will use particleFlow_cff, the new standard reference file.
Perhaps I need to be otherwise educated on how you pick a file for parsing

Reply:
@slava77 
If there is a cfi it picks that, rather than a cff, by default. What is in the cff and not cfi?

Reply:
era-related modifications are in the cff

Reply:
Era-dependent modifications used to be in cfi, but i moved them to cff, because now cfi is auto-generated, and I didn't know how to keep the era-related modifications in cfi which is now auto-generated.

Reply:
@Martin-Grunewald 
to expand on my note in https://github.com/cms-sw/cmssw/pull/28110#discussion_r336455010: please clarify if there is a specific rule in the HLT confDB parsing which requires using the auto-generated cfi file (which is incomplete for offline use case) rather than the partly manually made cff file (which is fully usable for offline configs).
In other words, is there anything that would prevent you from using the particleFlow_cff?

Reply:
@slava77 
Currently ConfDB parsing uses cfi by default if there is one, but I can tell it to use a cff. In this case, does the cfi simply contain less parameters, with the rest added by hand elsewhere or by fillDescriptions of other plugings/code?

Reply:
in this case it should be the same parameters, just the values can change depending on the era/modifier, which is defined/applied in the cff.

Is there some magic memory from the past in the cfi file discovery mechanism? If we rename this one to particleFlowDefault_cfi (as proposed above), would ConfDb still be able to find it?

Reply:
Yes, sure, it looks at any cfi it finds. Also, it is really more about the parameters themselves, not so much their precise values which may need adjustment for hlt anyway. So the name of the file could remain as is.

Reply:
ok. thx for clarification.